src/lib/fs_linux.cc

```diff
@@ -0,0 +1,41 @@
+/********************************************************************
+ * livegrep -- fs_linux.cc
+ * Copyright (c) 2011-2013 Nelson Elhage
+ *
+ * This program is free software. You may use, redistribute, and/or
+ * modify it under the terms listed in the COPYING file.
+ ********************************************************************/
+#include "fs.h"
+
+#include <limits.h>
+#include <sys/inotify.h>
+#include <unistd.h>
+
+namespace {
+    int fd = -1;
+    int wd = -1;
+}
+
+fswatcher::fswatcher(const std::string &path) : path_(path) {
+    fd = inotify_init();
+    wd = inotify_add_watch(fd, path.c_str(), IN_ATTRIB | IN_CLOSE_WRITE | IN_MOVE_SELF);
+}
+
+fswatcher::~fswatcher() {
+    if (fd >= 0) {
+        close(fd);
+        if (wd >= 0) {
+            inotify_rm_watch(fd, wd);
```

Comment:
Removing a watch _after_ we close the `fd` doesn't seem right. I think we don't need to clean up the watch at all if we're closing the `inotify` fd entirely, if I read `inotify(7)` right.